core/che-core-db/src/main/java/org/eclipse/che/core/db/JNDIDataSourceFactory.java

```diff
@@ -0,0 +1,55 @@
+/*******************************************************************************
+ * Copyright (c) 2012-2017 Red Hat, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.core.db;
+
+
+import org.apache.tomcat.dbcp.dbcp2.BasicDataSource;
+import org.apache.tomcat.dbcp.dbcp2.BasicDataSourceFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.naming.Context;
+import javax.naming.Name;
+import javax.naming.spi.ObjectFactory;
+import java.util.Hashtable;
+import java.util.Properties;
+
+public class JNDIDataSourceFactory implements ObjectFactory {
+
+    private static final Logger LOG = LoggerFactory.getLogger(JNDIDataSourceFactory.class);
+
+    private final BasicDataSource dataSource;
+
+    public JNDIDataSourceFactory(String userName,
+                                 String password,
+                                 String url,
+                                 String driverClassName,
+                                 String maxTotal,
+                                 String maxIdle,
+                                 String maxWaitMillis) throws Exception {
+        Properties poolConfigurationProperties = new Properties();
+        poolConfigurationProperties.setProperty("username", userName);
+        poolConfigurationProperties.setProperty("password", password);
+        poolConfigurationProperties.setProperty("url", url);
+        poolConfigurationProperties.setProperty("driverClassName", driverClassName);
+        poolConfigurationProperties.setProperty("maxTotal", maxTotal);
+        poolConfigurationProperties.setProperty("maxIdle", maxIdle);
+        poolConfigurationProperties.setProperty("maxWaitMillis", maxWaitMillis);
+        dataSource = BasicDataSourceFactory.createDataSource(poolConfigurationProperties);
+    }
+
+
+    @Override
+    public Object getObjectInstance(Object obj, Name name, Context nameCtx, Hashtable<?, ?> environment) throws Exception {
+        LOG.info("This={} obj={} name={} Context={} environment={}", this, obj, name, nameCtx, environment);
```

Comment:
https://github.com/eclipse/che/blob/che-multiuser/core/che-core-db/src/main/java/org/eclipse/che/core/db/JNDIDataSourceFactory.java#L59

It has been merged but it's still INFO level